scripts/hooks/.authenticate.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/sh
+# This script checks if the $HOOK_token environment variable is equal to the $DEFAULT_AUTH_TOKEN environment variable.
```

Comment:
When we do this in production, we'll use [Docker Secrets](https://docs.docker.com/engine/swarm/secrets/), which will mean the secret will be mounted in `/run/secrets/webhook_secret_token` (or a filename similar to that).  We'll have to compare the token we get to the value of this file.

You could probably do something [like this](https://stackoverflow.com/questions/11287861/how-to-check-if-a-file-contains-a-specific-string-using-bash).

Reply:
Okay I could do this with an order of precedence like this, ranging from highest priority to lowest.
1. token parameter/argument
2. Docker secret `DEFAULT_AUTH_TOKEN`
3. Container env var `DEFAULT_AUTH_TOKEN`

Also, If we want each script to have a different auth token, we can override the `DEFAULT_AUTH_TOKEN` secret/envVar by specifying the expected token as a parameter when calling the `.authenticate.sh` script, like `.authenticate.sh <providedToken> <expectedToken>`

Reply:
In a follow-up, we could also look at doing something where we generate a secret using a public/private key so we don't have to worry about rotating.  We don't need that here, but it would be more secure in future.